### PR TITLE
fix common_fnc_setAimCoef by making AF use it

### PR DESCRIPTION
**Pull request by nomisum**
**When merged this pull request will:**
- fix common_fnc_setAimCoef by making AF use it

### CODE REVIEW

**Review by jonpas — APPROVED** [review]

File: addons/advanced_fatigue/functions/fnc_handleEffects.sqf [hunk 1]
```diff
@@ -96,4 +96,4 @@ if (_overexhausted) then {
 _unit setVariable [QGVAR(aimFatigue), _fatigue];
 
 private _aimCoef = [missionNamespace, "ACE_setCustomAimCoef", "max"] call EFUNC(common,arithmeticGetResult);
-_unit setCustomAimCoef _aimCoef;
+[_unit, "ACE_setCustomAimCoef", _aimCoef, true] call EFUNC(common,setAimCoef);
```
Comment (BaerMitUmlaut) [inline addons/advanced_fatigue/functions/fnc_handleEffects.sqf:99]: "ACE_setCustomAimCoef" is not a unique ID for AF. Would recommend `QUOTE(ADDON)` or something similar.

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]

File: addons/advanced_fatigue/functions/fnc_handleEffects.sqf (hunk 1 above)
Comment (nomisum) [reply addons/advanced_fatigue/functions/fnc_handleEffects.sqf]: pushed your suggestion

**Comment by PabstMirror** [comment]
Medical and AdvFatigue both use
```
[missionNamespace, "ACE_setCustomAimCoef", QUOTE(ADDON), {
    //code that returns a number
}] call EFUNC(common,arithmeticSetSource);
```
so EFUNC(common,setAimCoef) is redundant as we already have the max

IMHO we should just drop setAimCoef as it's non-functional now
mod/mission makers can use "ACE_setCustomAimCoef"

we change it to be a wrapper like:
https://github.com/acemod/ACE3/blob/4669ec961d47eef78a14034590a569471366a5bf/addons/common/functions/fnc_setAimCoef.sqf

but the params don't line up  
- _unit is assumed to be player
- I don't like `compile str _setting`

**Comment by BaerMitUmlaut** [comment]
It's pretty bad how inconsistent we handle these things. `arithmeticSetSource` is a decent system, but not ideal either because one component still has the authority because at least one needs to use the actual engine function.

Ideally we would have a centralised system in `common` - something like the status effects system.

**Comment by PabstMirror** [comment]
We could just go back to using https://github.com/acemod/ACE3/blob/master/addons/common/functions/fnc_setAimCoef.sqf
it gets the max and applies it, seems to do everything we want

**Comment by jonpas** [comment]
@PabstMirror @BaerMitUmlaut status?

**Comment by Mike-MF** [comment]
@PabstMirror @BaerMitUmlaut Re-open if you want this.

Closing due to age and/or inactivity, unlikely to be finished or has to be reworked due to other changes over the years. May be re-opened if someone finds it useful, even better just open a new up-to-date PR.

**Comment by PabstMirror** [comment]
replaced with https://github.com/acemod/ACE3/pull/9226 I think

**Pull request closed without merging**